sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py

```diff
@@ -156,23 +167,24 @@ def resolve(self) -> None:
         for additional_include in self._includes:
             src_path = (base_path / additional_include).resolve()
             if self._is_folder_to_compress(src_path):
-                # note: additional include is a zip file, we need to compress corresponding folder,
-                # create a temp folder and copy files to this, filter __pycache__ in the folder
-                # during the copy.
-                # this operation can be replaced by using zipfile.ZipFile, which
-                # allows us pick files to compress rather than copy -- may improve performance.
-                zip_additional_include = (base_path / additional_include).resolve()
-                folder_to_zip = zip_additional_include.parent / zip_additional_include.stem
-                with tempfile.TemporaryDirectory() as tmp_dir:
-                    tmp_folder_to_zip = (Path(tmp_dir) / "zip").resolve()
-                    _copy_folder_ignore_pycache(folder_to_zip, tmp_folder_to_zip)
-                    src_path = (Path(tempfile.mkdtemp()) / zip_additional_include.name).resolve()
-                    shutil.make_archive(str(src_path.parent / src_path.stem), "zip", tmp_folder_to_zip)
-            dst_path = (tmp_folder_path / src_path.name).resolve()
-            self._copy(src_path, dst_path)
+                self._resolve_folder_to_compress(additional_include, Path(tmp_folder_path))
+            else:
+                dst_path = (tmp_folder_path / src_path.name).resolve()
+                self._copy(src_path, dst_path)
         self._tmp_code_path = tmp_folder_path  # point code path to tmp folder
         return
 
+    def _resolve_folder_to_compress(self, include: str, dst_path: Path) -> None:
+        zip_additional_include = (self._additional_includes_file_path.parent / include).resolve()
```

Comment:
should the following comments kept?

# note: additional include is a zip file, we need to compress corresponding folder,
# create a temp folder and copy files to this, filter __pycache__ in the folder
# during the copy.

Reply:
oh yes, this should be the docstring for this function, good catch!